src/client/InteractionClient.js

```diff
@@ -0,0 +1,232 @@
+      path = path.guilds(guildID);
+    }
+    const c = await path.commands.post({
+      data: transformCommand(command),
+    });
+    return new ApplicationCommand(this, c, guildID);
+  }
+
+  handle(data) {
+    switch (data.type) {
+      case InteractionType.PING:
+        return {
+          type: InteractionResponseType.PONG,
+        };
+      case InteractionType.APPLICATION_COMMAND: {
+        let timedOut = false;
+        let resolve;
+        const directPromise = new Promise(r => {
+          resolve = r;
+          this.client.setTimeout(() => {
+            timedOut = true;
+            r({
+              type: InteractionResponseType.DEFERRED_CHANNEL_MESSAGE_WITH_SOURCE,
+            });
+          }, 250);
+        });
+
+        const syncHandle = {
+          acknowledge() {
+            if (!timedOut) {
+              resolve({
+                type: InteractionResponseType.DEFERRED_CHANNEL_MESSAGE_WITH_SOURCE,
+              });
+            }
+          },
+          reply(resolved) {
+            if (timedOut) {
+              return false;
+            }
+            resolve({
+              type: InteractionResponseType.CHANNEL_MESSAGE_WITH_SOURCE,
+              data: resolved.data,
+            });
+            return true;
+          },
+        };
+
+        const interaction = new CommandInteraction(this.client, data, syncHandle);
+
+        /**
+         * Emitted when an interaction is created.
+         * @event Client#interactionCreate
+         * @param {Interaction} interaction The interaction which was created.
+         */
+        this.client.emit(Events.INTERACTION_CREATE, interaction);
+
+        return directPromise;
+      }
+      default:
+        throw new RangeError('Invalid interaction data');
+    }
+  }
+
+  /**
+   * An express-like middleware factory which can be used
+   * with webhook interactions.
+   * @returns {Function} The middleware function.
+   */
+  middleware() {
+    return async (req, res) => {
+      const timestamp = req.get('x-signature-timestamp');
+      const signature = req.get('x-signature-ed25519');
+
+      const chunks = [];
+      for await (const chunk of req) {
+        chunks.push(chunk);
+      }
+      const body = Buffer.concat(chunks);
+
+      if (sodium === undefined) {
+        sodium = require('../util/Sodium');
+      }
+      if (
+        !sodium.methods.verify(
+          Buffer.from(signature, 'hex'),
+          Buffer.concat([Buffer.from(timestamp), body]),
+          this.publicKey,
+        )
+      ) {
+        res.status(401).end();
+        return;
+      }
+
+      const data = JSON.parse(body.toString());
+
+      const result = await this.handle(data);
+      res.status(200).end(JSON.stringify(result));
+    };
+  }
+
+  async handleFromGateway(data) {
+    const result = await this.handle(data);
+
+    await this.client.api.interactions(data.id, data.token).callback.post({
+      data: result,
+      query: { wait: true },
```

Comment:
This query parameter doesn't exist for this endpoint as far as I can tell. And from what Mason has said, won't be added in the near future.